src/app/qgisapp.cpp

```diff
@@ -2219,6 +2219,8 @@ QgisApp::~QgisApp()
 
   QgsGui::nativePlatformInterface()->cleanup();
 
+  disconnect( mMessageButton, &QAbstractButton::toggled, this, &QgisApp::toggleLogMessageIcon );
```

Comment:
How about just:

    delete mMessageButton;
    mMessageButton = nullptr;

instead? That'd match the other logic here, and would be slightly more future proof vs explicit disconnect calls (eg if other connections to this object cause issues)

Reply:
Right :+1: Done (but with `unique_ptr`)

Reply:
I think the previous version with an explicit delete was better -- we can't predict the order that the unique_ptr will be deleted in, and if the crash is happening somewhere after the explicit QgisApp destructor is called then the crash may re-arise.

(For future reference, QObjectUniquePtr is a better choice for storing parented QObject's in a smart pointer vs std::unique_ptr, as we don't want crashes when the QObject is deleted by its parent (which a unique_ptr will have no knowledge of))

Reply:
Done